vllm/config.py

```diff
@@ -766,19 +767,24 @@ def __init__(self,
                  num_lookahead_slots: int = 0,
                  delay_factor: float = 0.0,
                  enable_chunked_prefill: bool = False,
-                 embedding_mode: Optional[bool] = False,
-                 preemption_mode: Optional[str] = None) -> None:
+                 preemption_mode: Optional[str] = None,
+                 model_mode: ModelMode = ModelMode.DECODER) -> None:
         if max_num_batched_tokens is not None:
             self.max_num_batched_tokens = max_num_batched_tokens
         else:
             if enable_chunked_prefill:
                 # It is the values that have the best balance between ITL
                 # and TTFT on A100. Note it is not optimized for throughput.
                 self.max_num_batched_tokens = 512
-            elif embedding_mode:
+            elif model_mode == ModelMode.EMBEDDING:
                 # For embedding, choose specific value for higher throughput
                 self.max_num_batched_tokens = max(
                     max_model_len, _EMBEDDING_MODEL_MAX_NUM_BATCHED_TOKENS)
+            elif model_mode == ModelMode.SIMPLE:
+                # For non-[decoder,embedding] model, choose specific value
+                # for higher throughput
+                self.max_num_batched_tokens = max(
+                    max_model_len, _SIMPLE_MODEL_MAX_NUM_BATCHED_TOKENS)
```

Comment:
Can also factor this out to `ModelMode`. (Return `_EMBEDDING_MODEL_MAX_NUM_BATCHED_TOKENS` etc. inside `ModelMode`, but keep the `max(...)` logic here.)

Reply:
Please take a look :)